## Exploratory Data Analysis

In [43]:
import pandas as pd


df = pd.read_csv(
    r"C:\Code\Python\DA_Projects\ecommerce_customer_sales_analytics\dataset\data.csv",
    encoding="latin1")

df

,InvoiceNo,StockCode,Description,Quantity,InvoiceDate,UnitPrice,CustomerID,Country
0,536365,85123A,WHITE HANGING HEART T-LIGHT HOLDER,6,12/1/2010 8:26,2.55,17850.0,United Kingdom
1,536365,71053,WHITE METAL LANTERN,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
2,536365,84406B,CREAM CUPID HEARTS COAT HANGER,8,12/1/2010 8:26,2.75,17850.0,United Kingdom
3,536365,84029G,KNITTED UNION FLAG HOT WATER BOTTLE,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
4,536365,84029E,RED WOOLLY HOTTIE WHITE HEART.,6,12/1/2010 8:26,3.39,17850.0,United Kingdom
...,...,...,...,...,...,...,...,...
541904,581587,22613,PACK OF 20 SPACEBOY NAPKINS,12,12/9/2011 12:50,0.85,12680.0,France
541905,581587,22899,CHILDREN'S APRON DOLLY GIRL,6,12/9/2011 12:50,2.10,12680.0,France
541906,581587,23254,CHILDRENS CUTLERY DOLLY GIRL,4,12/9/2011 12:50,4.15,12680.0,France
541907,581587,23255,CHILDRENS CUTLERY CIRCUS PARADE,4,12/9/2011 12:50,4.15,12680.0,France


In [ ]:
print(df.dtypes) # Datentypen der Spalten herausfinden

InvoiceNo                 str
StockCode                 str
Description               str
Quantity                int64
InvoiceDate    datetime64[us]
UnitPrice             float64
CustomerID            float64
Country                   str
dtype: object


0         False
1         False
2         False
3         False
4         False
          ...  
541904    False
541905    False
541906    False
541907    False
541908    False
Name: Quantity, Length: 541909, dtype: bool

In [44]:
# change date from string to numpy date
df["InvoiceDate"] = pd.to_datetime(df["InvoiceDate"], format="%m/%d/%Y %H:%M")

print(df["InvoiceDate"].head())
print(type(df["InvoiceDate"].iloc[0]))

0   2010-12-01 08:26:00
1   2010-12-01 08:26:00
2   2010-12-01 08:26:00
3   2010-12-01 08:26:00
4   2010-12-01 08:26:00
Name: InvoiceDate, dtype: datetime64[us]
<class 'pandas.Timestamp'>


### how many empy fields?

In [46]:
df.isnull().mean() * 100

InvoiceNo       0.000000
StockCode       0.000000
Description     0.268311
Quantity        0.000000
InvoiceDate     0.000000
UnitPrice       0.000000
CustomerID     24.926694
Country         0.000000
dtype: float64

Around 25% of all orders have no CustomerID and 0.26% have no desciption.

### do we have duplicates rows in the df?

In [47]:
df.duplicated().sum()

np.int64(5268)

We have 5268 duplicated rows in the df.

### what time window does the df has his data from?

In [49]:
print(df["InvoiceDate"].min())

print(df["InvoiceDate"].max())


2010-12-01 08:26:00
2011-12-09 12:50:00


from 01.12.2010 to 09.12.2011

### 6. how many customers do we have?

In [ ]:
df["CustomerID"].nunique() # nunqiue() ignores NaN

4372

we have 4372 unqiue Customers in the DataSet

### 7. how many different products do we have?

In [64]:
df["StockCode"].nunique()

4070

there are 4070 different products in the dataset

In [65]:
### 8. how many different countries does it get delivered to?
df["Country"].nunique()

38

38 different countries